exercises/concept/paulas-palindromes/.docs/hints.md

```diff
@@ -0,0 +1,42 @@
+# Hints
+
+## General
+
+- Check out the [Elm Guide][elm-guide], which has a nice worked example that is similar.
+
+## 1. Define the Model and Msg types for the application
+
+- The `Model` contains the application's state - all the data that the application needs.
+- This application just needs a string to store the text in the Text Box.
+- It can be any type, but in any useful application it is always a [record][record].
+
+- The `Msg` type is a defines the messages that are passed to the `update` function, to trigger specific changes in the model.
+- This application only needs one change to the model - updating the model when the text in the Text Box changes.
+- It can be any type, but in any useful application it is always a [custom type][custom-type].
+
+## 2. Write the update function
+
+- In any useful application the update function will use a `case` statement to pattern match on the `Msg` parameter.
+- In each branch of the case statement it will extract information it needs from the `Msg` parameter and return an updated `Model`
+- The `Model` should be a record, and [record update syntax][record-update-syntax] is normally used.
+
+## 3. Write the view function
+
+- The `view` function should probably return a `div` for the root element
+- The first child should be an `input`, with a `value` attribute for the current text, and an `onInput` attribute / event with the relevant variant of the `Msg`.
+- The second child should probably be another `div` with a `text` child stating whether the text is a palindrome or not.
+
+## 4. Write the init function
+
+- This should simply return a `Model` with sensible a default value (the tests require an empty string).
+
+## 5. Write the main function
+
+- The main function should just call [`Browser.sandbox`][browser-sandbox]
+- `Browser.sandbox` requires a [record][record] argument with the `init`, `update` and `view` functions.
```

Comment:
The type signature also needs to go from `Html msg` to `Html Msg`, so either we remove the type signature, either we add a hint about it.
I know that `msg` versus `Msg` is a big source of confusion for beginners so I'm leaning towards removing the type signature.

EDIT: oh, and later on when I tried to run the exercise locally, the signature clashed with `Program () Model Msg` so yes, let's just remove it.